pkg/quack/quack.go

```diff
@@ -181,21 +183,27 @@ func createPatch(old []byte, new []byte) ([]byte, error) {
 }
 
 func getTemplateInput(data []byte) ([]byte, error) {
-	// Remove annotations from input template
-	removeAnnotations := []byte(`[
-		{"op": "remove", "path": "/metadata/annotations"}
-	]`)
-	patch, err := mergepatch.DecodePatch(removeAnnotations)
+	// Fetch object meta into object
+	objectMeta, err := getObjectMeta(data)
 	if err != nil {
-		return []byte{}, fmt.Errorf("unable to decode remove annotation patch: %v", err)
+		return nil, fmt.Errorf("error reading object metadata: %v", err)
 	}
 
-	// Apply patch to remove annotations
-	templateInput, err := patch.Apply(data)
-	if err != nil {
-		return []byte{}, fmt.Errorf("unable to apply remove annotation patch: %v", err)
+	var patchedData []byte
+	for annotation := range objectMeta.Annotations {
+		if strings.HasPrefix(annotation, "quack.pusher.com") {
+			// Remove annotations from input template
+			patch := []byte(fmt.Sprintf(`[
+				{"op": "remove", "path": "/metadata/annotations/%s"}
+			]`, strings.Replace(annotation, "/", "~1", -1)))
```

Comment:
Can we break this up into two statements? I find it very hard to read and understand what's going on :)